Design/MoreFlexibleILiveListener.md

```diff
@@ -63,34 +64,89 @@ The general problem we are trying to solve is that the instrument is assumed to
 
 **In this solution users would be able to use `StartLiveData` and completely bypass the Facilities.xml file.** Again, this can be added in parallel to the above fixes.
+
+
+Proposed Solution
+-----------------
+
+#### Requirement 1
+
+Dynamic properties for listeners were already supported previously, but this feature was removed in [PR #234](https://github.com/mantidproject/mantid/pull/234) in response to [Trac #11059](http://trac.mantidproject.org/mantid/ticket/11059) because it interfered with help generation and Python API support.
+
+Reverting this PR will satisfy this requirement, but we will need to deal with the help and Python issues in a different way. This is beyond the scope of this design and can be handled separately.
+
+#### Requirement 2
+
+Support for multiple LiveListeners can be implemented using Option 1, described above.
+
+This will require changes to the way existing code and interfaces work. Facilities.xml should also be updated to replace workaround constructs like:
+
+```xml
+  <instrument name="ENGIN-X" shortname="ENGINX" >
+    <zeropadding size="8"/>
+    <technique>Neutron Diffraction</technique>
+    <livedata address="NDXENGINX:6789" />
+  </instrument>
+
+  <instrument name="ENGIN-X_EVENT" shortname="ENGINX" >
+    <zeropadding size="8"/>
+    <technique>Neutron Diffraction</technique>
+    <livedata address="NDXENGINX:10000" listener="ISISLiveEventDataListener" />
+  </instrument>
+```
+
+With a single instrument that has two connection types:
+
+```xml
+  <instrument name="ENGIN-X" shortname="ENGINX" >
+    <zeropadding size="8"/>
+    <technique>Neutron Diffraction</technique>
+    <livedata default="histo">
+      <connection name="histo" address="NDXENGINX:6789" listener="ISISHistoDataListener" />
+      <connection name="event" address="NDXENGINX:10000" listener="ISISLiveEventDataListener" />
+    </livedata>
+  </instrument>
+```
+
+#### Requirement 3
+
+Supporting direct LiveListener instantiation using a class name and connection string will require either modifying the existing `LiveListenerFactoryImpl::create` method, as per Option 2 above, or overloading it. The proposed solution is to overload it to provide an alternative while minimizing impact on existing code.
+
+The existing `create` method can be modified to function the same way it does now, but internally call the new `create` method, passing in the class name and address retrieved from Facilities.xml. `LiveDataAlgorithm` and `StartLiveData` (as well as its dialog) will need to be modified as well, to provide a user interface and call LiveListenerFactory accordingly.
+
+By adding a "Connection" group under the "Instrument" drop down box, to select an address / listener pair defined in the Facilities.xml using a drop down box, or to specify a custom address and listener, we could support both the general use case and allow for custom setups.
```

Comment:
Yes you're right with some of this. It always catches me out that the items in the `DynamicFactory` are their own factory. We really need a solution that doesn't require modifying the `facilities.xml` file. I haven't considered your proposal thoroughly yet though.